doc/examples/contextual_enhancement.py

```diff
@@ -0,0 +1,278 @@
+
+# Perform convolution
+csd_shm_enh = convolve(csd_shm_noisy, k, sh_order=8)
+
+
+"""
+The Spherical Deconvolution Transform is applied to sharpen the ODF field. 
+"""
+
+# Sharpen via the Spherical Deconvolution Transform
+from dipy.reconst.csdeconv import odf_sh_to_sharp
+csd_shm_enh_sharp = odf_sh_to_sharp(csd_shm_enh, sphere,  sh_order=8, lambda_=0.1)
+
+# Convert raw and enhanced data to discrete form
+csd_sf_orig = sh_to_sf(csd_shm_orig, sphere, sh_order=8)
+csd_sf_noisy = sh_to_sf(csd_shm_noisy, sphere, sh_order=8)
+csd_sf_enh = sh_to_sf(csd_shm_enh, sphere, sh_order=8)
+csd_sf_enh_sharp = sh_to_sf(csd_shm_enh_sharp, sphere, sh_order=8)
+
+# Normalize the sharpened ODFs
+csd_sf_enh_sharp = csd_sf_enh_sharp * np.amax(csd_sf_orig)/np.amax(csd_sf_enh_sharp)
+
+""" 
+The end results are visualized. It can be observed that the end result after 
+diffusion and sharpening is closer to the original noiseless dataset.
+"""
+
+csd_sf_orig_slice = csd_sf_orig[:,:,[3],:]
+csd_sf_noisy_slice = csd_sf_noisy[:,:,[3],:]
+csd_sf_enh_slice = csd_sf_enh[:,:,[3],:]
+csd_sf_enh_sharp_slice = csd_sf_enh_sharp[:,:,[3],:]
+
+ren = fvtk.ren()
+
+# original ODF field
+fodf_spheres_org = fvtk.sphere_funcs(csd_sf_orig_slice, 
+                                     sphere,
+                                     scale=2, 
+                                     norm=False, 
+                                     radial_scale=True)
+fodf_spheres_org.SetPosition(0, 35, 0)
+fvtk.add(ren, fodf_spheres_org)
+
+# ODF field with added noise
+fodf_spheres = fvtk.sphere_funcs(csd_sf_noisy_slice, 
+                                 sphere, 
+                                 scale=2, 
+                                 norm=False, 
+                                 radial_scale=True)
+fodf_spheres.SetPosition(0, 0, 0)
+fvtk.add(ren, fodf_spheres)
+
+# Enhancement of noisy ODF field
+fodf_spheres_enh = fvtk.sphere_funcs(csd_sf_enh_slice, 
+                                     sphere, 
+                                     scale=2, 
+                                     norm=False, 
+                                     radial_scale=True)
+fodf_spheres_enh.SetPosition(35, 0, 0)
+fvtk.add(ren, fodf_spheres_enh)
+
+# Additional sharpening
+fodf_spheres_enh_sharp = fvtk.sphere_funcs(csd_sf_enh_sharp_slice * 1.5, 
+                                           sphere, 
+                                           scale=2, 
+                                           norm=False, 
+                                           radial_scale=True)
+fodf_spheres_enh_sharp.SetPosition(35, 35, 0)
+fvtk.add(ren, fodf_spheres_enh_sharp)
+
+fvtk.record(ren, out_path='enhancements.png', size=(900, 900))
+
+"""
+
+.. figure:: enhancements.png
+   :align: center
+
+   The results after enhancements. Top-left: original noiseless data. Bottom-left: 
+   original data with added Rician noise (SNR=2). Bottom-right: After enhancement 
+   of noisy data. Top-right: After enhancement and sharpening of noisy data.
+
```

Comment:
Interesting results! I just ran this example, which I find to be great - both comprehensive in terms of explaining the method, and also nice clean code to demonstrate how it's used.